File: БР1.2/Smirnova_Karina/labs/lab1/src/main/java/org/renting/rentingservice/exception/ApiException.java
package org.renting.rentingservice.exception;

import lombok.Getter;
import org.springframework.http.HttpStatus;

import java.util.Map;

@Getter
public class ApiException extends RuntimeException {

    private final String code;
    private final HttpStatus status;
    private final Map<String, Object> details;

    public ApiException(String code, String message, HttpStatus status) {
        this(code, message, status, null);
    }

    public ApiException(String code, String message, HttpStatus status, Map<String, Object> details) {
        super(message);
        this.code = code;
        this.status = status;
        this.details = details;
    }
}

File: БР1.2/Цатинян Артём/labs/lab2/common/src/main/kotlin/ru/itmo/restaurantbooking/lab2/common/exception/ApiException.kt
package ru.itmo.restaurantbooking.lab2.common.exception

import org.springframework.http.HttpStatus

open class ApiException(
    val status: HttpStatus,
    override val message: String
) : RuntimeException(message)

class BadRequestException(message: String) : ApiException(HttpStatus.BAD_REQUEST, message)

class UnauthorizedException(message: String) : ApiException(HttpStatus.UNAUTHORIZED, message)

class ForbiddenException(message: String) : ApiException(HttpStatus.FORBIDDEN, message)

class NotFoundException(message: String) : ApiException(HttpStatus.NOT_FOUND, message)

class ConflictException(message: String) : ApiException(HttpStatus.CONFLICT, message)

class UnprocessableEntityException(message: String) : ApiException(HttpStatus.UNPROCESSABLE_ENTITY, message)
